hiop@1.0.0 +cuda ^cuda@12.0.1: LinAlg/hiopMatrixSparseCsrCuda.cpp:991:38: error: 'CUSPARSE_CSR2CSC_ALG2' was not declared in this scope; did you mean 'CUSPARSE_CSR2CSC_ALG1'?

### Steps to reproduce the issue

`hiop@1.0.0 +cuda cuda_arch=80` build fails using:
* `spack@develop` (1c0d3bc07158578a34ea5fd4f7565386085302c0 from `Sun Oct 22 22:11:19 2023 +0200`)
* CUDA 12.0.1
* GCC 11.4.0
* Ubuntu 20.04, X86_64

Spack environment:
```
spack:
  view: false

  concretizer:
    reuse: false
    unify: false

  compilers:
  - compiler:
      spec: gcc@=11.4.0
      paths:
        cc: /usr/bin/gcc
        cxx: /usr/bin/g++
        f77: /usr/bin/gfortran
        fc: /usr/bin/gfortran
      flags: {}
      operating_system: ubuntu20.04
      target: x86_64
      modules: []
      environment: {}
      extra_rpaths: []

  packages:
    all:
      require: '%gcc'
      providers:
        blas: [openblas]
        mpi: [mpich]
      target: [x86_64]
      variants: +mpi

    boost:
      require: cxxstd=17 
    libffi:
      require: "@3.4.4"

    # MPI
    mpi:
      require: mpich
    mpich:
      require: '@4.1.2 ~wrapperrpath ~hwloc'
      buildable: false
      externals:
      - spec: mpich@4.1.2 ~argobots~cuda+fortran~hwloc+hydra+libxml2+pci~rocm+romio~slurm~vci~verbs~wrapperrpath build_system=autotools datatype-engine=auto device=ch4 netmod=ofi pmi=pmi
        prefix: /usr/local/mpich/install/mpich
    cuda:
      require: "@12.0.1"

  specs:
  - exago@1.5.1 +mpi +python +raja +hiop ~rocm +cuda cuda_arch=80 ~ipopt ^hiop@1.0.0 ~sparse +mpi +raja ~rocm +cuda cuda_arch=80 ^raja@0.14.0 ^python@3.10

```

<details><summary>Concretization</summary>

<pre>
==> Starting concretization
==> Environment concretized in 48.97 seconds
==> Concretized exago@1.5.1+cuda+hiop~ipopt+mpi+python+raja~rocm cuda_arch=80 ^hiop@1.0.0+cuda+mpi+raja~rocm~sparse cuda_arch=80 ^python@3.10 ^raja@0.14.0
 -   yatd3h6  exago@1.5.1%gcc@11.4.0+cuda+hiop~ipo~ipopt~logging+mpi+python+raja~rocm build_system=cmake build_type=Release cuda_arch=80 generator=make arch=linux-ubuntu20.04-x86_64
[+]  zj5unyh      ^camp@0.2.3%gcc@11.4.0+cuda~ipo+openmp~rocm~tests build_system=cmake build_type=Release cuda_arch=80 generator=make arch=linux-ubuntu20.04-x86_64
[+]  ddfzz5e          ^blt@0.4.1%gcc@11.4.0 build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  bhw3vkv          ^cub@2.1.0%gcc@11.4.0 build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  737xxmc      ^cmake@3.27.7%gcc@11.4.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=linux-ubuntu20.04-x86_64
[+]  vc43uif          ^curl@8.4.0%gcc@11.4.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=linux-ubuntu20.04-x86_64
[+]  k2d4pld              ^nghttp2@1.52.0%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  m7dbp7w          ^ncurses@6.4%gcc@11.4.0~symlinks+termlib abi=none build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  knn3iiz          ^zlib-ng@2.1.4%gcc@11.4.0+compat+opt build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  bhslxxm      ^cuda@12.0.1%gcc@11.4.0~allow-unsupported-compilers~dev build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  tj5lldx          ^libxml2@2.10.3%gcc@11.4.0+pic~python+shared build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  x4ev7kg              ^libiconv@1.17%gcc@11.4.0 build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  w7qwcrr      ^gmake@4.4.1%gcc@11.4.0~guile build_system=generic arch=linux-ubuntu20.04-x86_64
 -   v4c2ugv      ^hiop@1.0.0%gcc@11.4.0+cuda~cusolver_lu~deepchecking~ginkgo~ipo~jsrun~kron+mpi+raja~rocm~shared~sparse build_system=cmake build_type=Release cuda_arch=80 generator=make arch=linux-ubuntu20.04-x86_64
[+]  dzm524j      ^magma@2.7.2%gcc@11.4.0+cuda+fortran~ipo~rocm+shared build_system=cmake build_type=Release cuda_arch=80 generator=make arch=linux-ubuntu20.04-x86_64
[e]  4vf2w2d      ^mpich@4.1.2%gcc@11.4.0~argobots~cuda+fortran~hwloc+hydra+libxml2+pci~rocm+romio~slurm~vci~verbs~wrapperrpath build_system=autotools datatype-engine=auto device=ch4 netmod=ofi pmi=pmi arch=linux-ubuntu20.04-x86_64
[+]  4pmvb4g      ^openblas@0.3.24%gcc@11.4.0~bignuma~consistent_fpcsr+fortran~ilp64+locking+pic+shared build_system=makefile symbol_suffix=none threads=none arch=linux-ubuntu20.04-x86_64
[+]  opk2fdh          ^perl@5.38.0%gcc@11.4.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-ubuntu20.04-x86_64
[+]  bvpocrz              ^berkeley-db@18.1.40%gcc@11.4.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-ubuntu20.04-x86_64
[+]  jmcdptd      ^petsc@3.19.6%gcc@11.4.0~X~batch~cgns~complex~cuda~debug+double~exodusii~fftw+fortran~giflib+hdf5~hpddm~hwloc+hypre~int64~jpeg~knl~kokkos~libpng~libyaml~memkind+metis~mkl-pardiso~mmg~moab~mpfr+mpi~mumps~openmp~p4est~parmmg~ptscotch~random123~rocm~saws~scalapack+shared~strumpack~suite-sparse+superlu-dist~sycl~tetgen~trilinos~valgrind build_system=generic clanguage=C memalign=none arch=linux-ubuntu20.04-x86_64
[+]  mfkxhgk          ^diffutils@3.9%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  qhpe4ac          ^hdf5@1.14.2%gcc@11.4.0~cxx~fortran~hl~ipo~java~map+mpi+shared~szip~threadsafe+tools api=default build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-x86_64
[+]  i7ugfcc          ^hypre@2.28.0%gcc@11.4.0~caliper~complex~cuda~debug+fortran~gptune~int64~internal-superlu~magma~mixedint+mpi~openmp~rocm+shared~superlu-dist~sycl~umpire~unified-memory build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  r2uvvqx          ^metis@5.1.0%gcc@11.4.0~gdb~int64~ipo~real64+shared build_system=cmake build_type=Release generator=make patches=4991da9,93a7903,b1225da arch=linux-ubuntu20.04-x86_64
[+]  v67x4a4          ^parmetis@4.0.3%gcc@11.4.0~gdb~int64~ipo+shared build_system=cmake build_type=Release generator=make patches=4f89253,50ed208,704b84f arch=linux-ubuntu20.04-x86_64
[+]  ac5fmbq          ^superlu-dist@8.1.2%gcc@11.4.0~cuda~int64~ipo~openmp~rocm+shared build_system=cmake build_type=Release generator=make arch=linux-ubuntu20.04-x86_64
[+]  x6ksnrs      ^pkgconf@1.9.5%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  m4fx3a2      ^py-mpi4py@3.1.4%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  d6mhhiz          ^py-cython@0.29.36%gcc@11.4.0 build_system=python_pip patches=c4369ad arch=linux-ubuntu20.04-x86_64
[+]  w724moh          ^py-pip@23.1.2%gcc@11.4.0 build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  j3n5ffn          ^py-setuptools@68.0.0%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  s3bocfh          ^py-wheel@0.41.2%gcc@11.4.0 build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  dr6hctk      ^py-pytest@7.3.2%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  sv3niar          ^py-exceptiongroup@1.1.1%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  k2svzfe              ^py-flit-scm@1.7.0%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  ffe6oim          ^py-iniconfig@2.0.0%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  c2toa7j              ^py-hatch-vcs@0.3.0%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  4glbvgt              ^py-hatchling@1.18.0%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  sgxeske                  ^py-editables@0.3%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  45dsinu                  ^py-pathspec@0.11.1%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  ee3uzfx                  ^py-trove-classifiers@2023.8.7%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  5s5rbax                      ^py-calver@2022.6.26%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  ikvwyj5          ^py-packaging@23.1%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  qabtgfi              ^py-flit-core@3.9.0%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  edwocx3          ^py-pluggy@1.0.0%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  43lepqa          ^py-setuptools-scm@7.1.0%gcc@11.4.0+toml build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  ou6cwyt              ^git@2.42.0%gcc@11.4.0+man+nls+perl+subtree~svn~tcltk build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  j4iolkh                  ^autoconf@2.69%gcc@11.4.0 build_system=autotools patches=35c4492,7793209,a49dd5b arch=linux-ubuntu20.04-x86_64
[+]  dhc5ftf                  ^automake@1.16.5%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  vdcvema                  ^libidn2@2.3.4%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  7hvjwle                      ^libunistring@1.1%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  qp7l7qc                  ^libtool@2.4.7%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  2zk6ze6                  ^m4@1.4.19%gcc@11.4.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-ubuntu20.04-x86_64
[+]  owwhrha                      ^libsigsegv@2.14%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  7dkgnsg                  ^openssh@9.5p1%gcc@11.4.0+gssapi build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  srgv72c                      ^krb5@1.20.1%gcc@11.4.0+shared build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  7jlz6cy                          ^bison@3.8.2%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  53tnglq                          ^findutils@4.9.0%gcc@11.4.0 build_system=autotools patches=440b954 arch=linux-ubuntu20.04-x86_64
[+]  ziyvxx3                      ^libedit@3.1-20210216%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  klc6ljw                  ^pcre2@10.42%gcc@11.4.0~jit+multibyte build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  imz6ksq              ^py-typing-extensions@4.8.0%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  cszafo3          ^py-tomli@2.0.1%gcc@11.4.0 build_system=python_pip arch=linux-ubuntu20.04-x86_64
[+]  44ledya      ^python@3.10.13%gcc@11.4.0+bz2+crypt+ctypes+dbm~debug+libxml2+lzma~nis~optimizations+pic+pyexpat+pythoncmd+readline+shared+sqlite3+ssl~tkinter+uuid+zlib build_system=generic patches=0d98e93,7d40923,ebdca64,f2fd060 arch=linux-ubuntu20.04-x86_64
[+]  rz3egrw          ^bzip2@1.0.8%gcc@11.4.0~debug~pic+shared build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  qvhso2r          ^expat@2.5.0%gcc@11.4.0+libbsd build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  y5cb7vy              ^libbsd@0.11.7%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  wzkdojx                  ^libmd@1.0.4%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  x75cg2o          ^gdbm@1.23%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  rdsmzah          ^gettext@0.22.3%gcc@11.4.0+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  x3qw7ht              ^tar@1.34%gcc@11.4.0 build_system=autotools zip=pigz arch=linux-ubuntu20.04-x86_64
[+]  u7gw2ye                  ^pigz@2.7%gcc@11.4.0 build_system=makefile arch=linux-ubuntu20.04-x86_64
[+]  qjffzez                  ^zstd@1.5.5%gcc@11.4.0+programs build_system=makefile compression=none libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  wzudco5          ^libffi@3.4.4%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  k3vz6yk          ^libxcrypt@4.4.35%gcc@11.4.0~obsolete_api build_system=autotools patches=4885da3 arch=linux-ubuntu20.04-x86_64
[+]  322sfmt          ^openssl@3.1.3%gcc@11.4.0~docs+shared build_system=generic certs=mozilla arch=linux-ubuntu20.04-x86_64
[+]  thgudgh              ^ca-certificates-mozilla@2023-05-30%gcc@11.4.0 build_system=generic arch=linux-ubuntu20.04-x86_64
[+]  2gxmxfz          ^readline@8.2%gcc@11.4.0 build_system=autotools patches=bbf97f1 arch=linux-ubuntu20.04-x86_64
[+]  nnoaq2z          ^sqlite@3.43.2%gcc@11.4.0+column_metadata+dynamic_extensions+fts~functions+rtree build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  sodcarg          ^util-linux-uuid@2.38.1%gcc@11.4.0 build_system=autotools arch=linux-ubuntu20.04-x86_64
[+]  naalujk          ^xz@5.4.1%gcc@11.4.0~pic build_system=autotools libs=shared,static arch=linux-ubuntu20.04-x86_64
[+]  o4xljdg      ^raja@0.14.0%gcc@11.4.0+cuda+examples+exercises~ipo+openmp~rocm+shared~tests build_system=cmake build_type=Release cuda_arch=80 generator=make arch=linux-ubuntu20.04-x86_64
[+]  qvzhm3g      ^umpire@6.0.0%gcc@11.4.0+c+cuda~device_alloc~deviceconst+examples~fortran~ipo~numa~openmp~rocm~shared build_system=cmake build_type=Release cuda_arch=80 generator=make tests=none arch=linux-ubuntu20.04-x86_64


</pre></details>


Reproduced using the Spack environment above and Docker container image:
`ecpe4s/ubuntu20.04-runner-amd64-gcc-11.4-rocm5.4.3-mpi-base:2023.08.20`:
```
$> spack -e . install
...
==> Installing hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23 [49/51]
==> No binary for hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23 found: installing from source
==> Using cached archive: /spack/var/spack/cache/_source-cache/git//LLNL/hiop.git/10b7d3ee0a15cb4949ccee8c905d447b9528794f.tar.gz
==> No patches needed for hiop
==> Warning: Fetching from mirror without a checksum!
  This package is normally checked out from a version control system, but it has been archived on a spack mirror.  This means we cannot know a checksum for the tarball in advance. Be sure that your connection to this mirror is secure!
==> hiop: Executing phase: 'cmake'
==> hiop: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j32'

5 errors found in build log:
     201    cd /tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-build-v4c2ugv/src/LinAlg && /spack/lib/spack/env/gcc/g++ -DCAMP_HAVE_CUDA -I/tmp/root/spack-stage
            /spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-build-v4c2ugv -I/tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-src/src/Interface -I/
            tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-src/src/ExecBackends -I/tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/
            spack-src/src/Optimization -I/tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-src/src/LinAlg -I/tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk6
            3q2uvhcfyo4fvifq5taag23/spack-src/src/Utils -I/tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-src/src/_Internals -I/spack/opt/spack/linux-ubuntu20.0
            4-x86_64/gcc-11.4.0/magma-2.7.2-dzm524jl6k2xndoq5rse7fg5yzxaccm7/include -isystem /usr/local/mpich/install/mpich/include -isystem /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/c
            uda-12.0.1-bhslxxm6co7nboucmtdphkt3uowxtyp2/include -isystem /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/camp-0.2.3-zj5unyhz3aj5zhijy6u3vgvgskpvtgxt/include -isystem /spack/op
            t/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/umpire-6.0.0-qvzhm3gxzghur4wytrbt3r7zk25r7qzb/include -isystem /spack/opt/spack/linux-ubuntu20.04-x86_64/gcc-11.4.0/raja-0.14.0-o4xljdgwpjws
            eovccwlkwo6qsem5o3oz/include -O3 -DNDEBUG -std=gnu++14 -fPIC -fopenmp -fopenmp -MD -MT src/LinAlg/CMakeFiles/hiopLinAlg.dir/hiopLinSolverSymDenseMagma.cpp.o -MF CMakeFiles/hiopLinAlg.d
            ir/hiopLinSolverSymDenseMagma.cpp.o.d -o CMakeFiles/hiopLinAlg.dir/hiopLinSolverSymDenseMagma.cpp.o -c /tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spa
            ck-src/src/LinAlg/hiopLinSolverSymDenseMagma.cpp
     202    /tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-src/src/LinAlg/hiopVectorCompoundPD.cpp: In member function 'virtual bool hiop::hiopVectorCompoundPD
            ::is_equal(const hiop::hiopVector&) const':
     203    /tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-src/src/LinAlg/hiopVectorCompoundPD.cpp:1010:1: warning: no return statement in function returning n
            on-void [-Wreturn-type]
     204     1010 | }
     205          | ^
     206    /tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-src/src/LinAlg/hiopMatrixSparseCsrCuda.cpp: In member function 'virtual void hiop::hiopMatrixSparseC
            SRCUDA::form_transpose_from_symbolic(const hiop::hiopMatrixSparseCSR&)':
  >> 207    /tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-src/src/LinAlg/hiopMatrixSparseCsrCuda.cpp:991:38: error: 'CUSPARSE_CSR2CSC_ALG2' was not declared i
            n this scope; did you mean 'CUSPARSE_CSR2CSC_ALG1'?
     208      991 |                                      CUSPARSE_CSR2CSC_ALG2,
     209          |                                      ^~~~~~~~~~~~~~~~~~~~~
     210          |                                      CUSPARSE_CSR2CSC_ALG1
...
  >> 212    /tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-src/src/LinAlg/hiopMatrixSparseCsrCuda.cpp:1020:27: error: 'CUSPARSE_CSR2CSC_ALG2' was not declared
            in this scope; did you mean 'CUSPARSE_CSR2CSC_ALG1'?
     213     1020 |                           CUSPARSE_CSR2CSC_ALG2,
     214          |                           ^~~~~~~~~~~~~~~~~~~~~
     215          |                           CUSPARSE_CSR2CSC_ALG1
  >> 216    make[2]: *** [src/LinAlg/CMakeFiles/hiopLinAlg.dir/build.make:289: src/LinAlg/CMakeFiles/hiopLinAlg.dir/hiopMatrixSparseCsrCuda.cpp.o] Error 1
     217    make[2]: *** Waiting for unfinished jobs....
     218    /tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-src/src/LinAlg/hiopVectorCuda.cpp: In member function 'virtual bool hiop::hiopVectorCuda::is_equal(c
            onst hiop::hiopVector&) const':
     219    /tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-src/src/LinAlg/hiopVectorCuda.cpp:1088:1: warning: no return statement in function returning non-voi
            d [-Wreturn-type]
     220     1088 | }
     221          | ^
     222    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-build-v4c2ugv'

     ...

     233
     234    /tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-src/src/LinAlg/hiopVectorRajaImpl.hpp(1219): warning #177-D: variable "tmp_n_local" was declared but
             never referenced
     235
     236    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-build-v4c2ugv'
     237    [ 42%] Built target hiopUtils
     238    make[2]: Leaving directory '/tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-build-v4c2ugv'
  >> 239    make[1]: *** [CMakeFiles/Makefile2:398: src/LinAlg/CMakeFiles/hiopLinAlg.dir/all] Error 2
     240    make[1]: Leaving directory '/tmp/root/spack-stage/spack-stage-hiop-1.0.0-v4c2ugvk63q2uvhcfyo4fvifq5taag23/spack-build-v4c2ugv'
  >> 241    make: *** [Makefile:149: all] Error 2
```

### Error message

<details><summary>Error message</summary>

<pre>
... see above
</pre></details>


### Information on your system

* **Spack:** 0.21.0.dev0 (1c0d3bc07158578a34ea5fd4f7565386085302c0)
* **Python:** 3.8.10
* **Platform:** linux-ubuntu20.04-cascadelake
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/13078718/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/13078719/spack-build-out.txt)

@cameronrutherford @pelesh @ryandanehy 

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

--- a/var/spack/repos/builtin/packages/hiop/package.py
+++ b/var/spack/repos/builtin/packages/hiop/package.py
@@ -103,10 +103,9 @@ class Hiop(CMakePackage, CudaPackage, ROCmPackage):
         depends_on("magma@{0}:".format(magma_v), when="@{0}:+cuda".format(hiop_v))
         depends_on("magma@{0}:".format(magma_v), when="@{0}:+rocm".format(hiop_v))
 
-    depends_on("cuda@11:", when="@develop:+cuda")
-
     # https://github.com/spack/spack/issues/40678
-    depends_on("cuda@:11.9", when="@:1.0 +cuda")
+    depends_on("cuda@11:11.9", when="@develop:+cuda")
+    depends_on("cuda@:11.9", when="+cuda")
 
     depends_on("raja", when="+raja")
     depends_on("umpire", when="+raja")

